x-pack/plugin/metering-ix/src/main/java/org/elasticsearch/xpack/meter/ix/MeterIXPoller.java

```diff
@@ -0,0 +1,153 @@
+    }
+
+    @Override
+    public void addLifecycleListener(LifecycleListener listener) {
+
+    }
+
+    @Override
+    public void start() {
+        mutex.lock();
+        try {
+            if (state == Lifecycle.State.INITIALIZED) {
+                state = Lifecycle.State.STARTED;
+                setScheduleNext();
+            }
+        } finally {
+            mutex.unlock();
+        }
+    }
+
+    @Override
+    public void stop() {
+        mutex.lock();
+        try {
+            if (state != Lifecycle.State.STOPPED && state != Lifecycle.State.CLOSED) {
+                state = Lifecycle.State.STOPPED;
+                if (next != null) {
+                    next.cancel();
+                }
+                next = null;
+            }
+        } finally {
+            mutex.unlock();
+        }
+    }
+
+    private void setScheduleNext() {
+        next = threadPool.schedule(this::logAndSchedule, TimeValue.timeValueSeconds(20), ThreadPool.Names.GENERIC);
+    }
+
+    public void logAndSchedule() {
+        logIndexStats();
+        mutex.lock();
+        try {
+            if (state == Lifecycle.State.STARTED) {
+                setScheduleNext();
+            }
+        } finally {
+            mutex.unlock();
+        }
+    }
+
+    public void logIndexStats() {
+        logger.warn("[IX] logIndexStats");
+        collectRecords();
+        logger.warn("[IX] done logIndexStats");
+    }
+
+    public List<IXRecord> collectRecords() {
+        List<IXRecord> records = new ArrayList<>();
+        for (final IndexService indexService : indicesService) {
+            String index = indexService.index().getName();
+            for (final IndexShard shard : indexService) {
+                Engine engine = shard.getEngineOrNull();
+                if (engine == null) {
+                    continue;
+                }
+                long term = shard.getOperationPrimaryTerm();
+                int shardId = shard.shardId().id();
+                for (final SegmentCommitInfo commitInfo : engine.getLastCommittedSegmentInfos()) {
+                    try {
+                        var record = new IXRecord(
+                            index,
+                            shardId,
+                            term,
+                            commitInfo.getDocValuesGen(),
+                            StringHelper.idToString(commitInfo.getId()),
+                            commitInfo.sizeInBytes()
+                        );
+                        records.add(record);
+                        logger.info("{}", record);
+                    } catch (IOException err) {
+                        logger.warn("error: [{}], index: [{}], shard: [{}], id: [{}]", err, index, shardId, commitInfo.getId());
+                    }
+                }
+            }
+        }
+        return records;
+    }
+
+    private record IXRecord(String index, int shardId, long term, long generation, String commitId, long size) {}
```

Comment:
What's the term, generation, and commitId recorded for?

Reply:
Removed in https://github.com/elastic/elasticsearch/pull/97978